fairseq/models/wav2vec/wav2vec2_scribblelens.py

```diff
@@ -525,6 +544,43 @@ def compute_preds(self, x, y, negatives):
 
         return logits
 
+    def smartpool(self, features, padding_mask=None):
+        features_tmp = F.pad(features,(0,0,1,0))
+        new_lens = (features_tmp[:,1:,:] - features_tmp[:,:-1,:]).abs().sum(dim=2)
+        new_lens = new_lens / new_lens.sum(1, keepdim=True) * (features.size(1) / self.smartpooling_factor) # Reducing the original length T by some factor
+        
+        features, interp_weights = self.warp(features, new_lens)
+        if padding_mask is not None :
```

Comment:
Padding mask oznacza które kolumny są poza danymi, więc paddingu nie skracamy, to co trzeba zrobić to skrócić te cześci unpadded, a potem sprawdzić który jest najdłuższy i spadować resztę.